bindings/go/input/dir/blob.go

```diff
@@ -0,0 +1,210 @@
+package dir
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"ocm.software/open-component-model/bindings/go/blob"
+	"ocm.software/open-component-model/bindings/go/blob/compression"
+	"ocm.software/open-component-model/bindings/go/blob/filesystem"
+	"ocm.software/open-component-model/bindings/go/blob/inmemory"
+	v1 "ocm.software/open-component-model/bindings/go/input/dir/spec/v1"
+)
+
+// GetV1DirBlob creates a ReadOnlyBlob from a v1.Dir specification.
+// It reads the directory from the filesystem and applies compression if requested.
+// The function returns an error if the file path is empty or if there are issues reading the directory
+// contents from the filesystem.
+//
+// The function performs the following steps:
+//  1. Validates that the directory path is not empty
+//  2. Reads the directory contents using an instance of the virtual FileSystem
+//  3. Packs the directory contents into a tar archive
+//  4. Applies different configuration options of the v1.Dir specification
+func GetV1DirBlob(dir v1.Dir) (blob.ReadOnlyBlob, error) {
+	if dir.Path == "" {
+		return nil, fmt.Errorf("dir path must not be empty")
+	}
+
+	// TODO:
+	// - Handle FollowSymlinks option.
+
+	// Pack directory contents as a tar archive.
+	reader, err := packDirToTar(dir.Path, &dir)
+	if err != nil {
+		return nil, fmt.Errorf("error producing blob for a dir input: %w", err)
+	}
+
+	// Wrap the tar archive in a ReadOnlyBlob.
+	var dirBlob blob.ReadOnlyBlob = inmemory.New(reader, inmemory.WithMediaType(dir.MediaType))
+
+	// gzip the blob, if requested in the spec.
+	if dir.Compress {
+		dirBlob = compression.Compress(dirBlob)
+	}
+
+	return dirBlob, nil
+}
+
+// packDirToTar is the main function, which creates a tar archive from the contents of the specified directory.
+// It creates an instance of the virtual FileSystem based on the directory path, creates a tar writer and
+// triggers recursive packaging of the directory contents.
+func packDirToTar(path string, dir *v1.Dir) (_ io.Reader, err error) {
+	// Determine the base directory for relative paths in the tar archive.
+	baseDir := path
+	if dir.PreserveDir {
+		// PreserveDir defines that the directory specified in the path field should be included in the blob.
+		baseDir = filepath.Dir(path)
+	}
+
+	// Create a new virtual FileSystem instance based on the provided directory path.
+	fs, err := filesystem.NewFS(baseDir, os.O_RDONLY)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create filesystem while trying to access %v: %w", path, err)
+	}
+
+	// Create a new tar writer.
+	var buf bytes.Buffer
+	tw := tar.NewWriter(&buf)
+	defer func() {
+		closeErr := tw.Close()
```

Comment:
you can use errors.Join here because the function already excludes nil errors passed.
```suggestion
		err = errors.Join(err, tw.Close())
```